Kernel/Arch/aarch64/ASM_wrapper.h

```diff
@@ -92,5 +92,13 @@ inline void enter_el1_from_el2()
                  "entered_el1:" ::
                      : "x0");
 }
+}
+
+namespace Kernel {
+inline bool are_interrupts_enabled()
+{
+    auto daif = Aarch64::DAIF::read();
+    return !daif.I;
+}
```

Comment:
I've been using `Processor::are_interrupts_enabled` instead, so you could replace the callsites of this function with `Processor::are_interrupts_enabled`, or call `Processor::are_interrupts_enabled()` in this function.

Reply:
👍